tests/src/rules/no-cycle.js

```diff
@@ -17,6 +18,10 @@ const testVersion = (specifier, t) => _testVersion(specifier, () => Object.assig
 
 const testDialects = ['es6'];
 
+beforeEach(() => {
+  rule._traversed.clear();
```

Comment:
this one, where without this hack, the *tests themselves* will fail, because the cache is lasting too long. This is the same failure that consumers will see.

One possible alternative is to add a `Program:exit` visitor, that clears the Set (after removing this line) - but someone would have to test that change and see if it still provided the same speedup.